evalml/automl/engine/sequential_engine.py

```diff
@@ -0,0 +1,28 @@
+from copy import copy
+
+from evalml.automl.engine import EngineBase
+
+
+class SequentialEngine(EngineBase):
+    """The default engine for the AutoML search. Trains and scores pipelines locally, one after another."""
+
+    def evaluate_batch(self, pipelines):
+        """Evaluate a batch of pipelines using the current dataset and AutoML state.
+
+        Arguments:
+            pipelines (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+
+        Returns:
+            list (int): a list of the new pipeline IDs which were created by the AutoML search.
+        """
+        if self.X_train is None or self.y_train is None:
+            raise ValueError("Dataset has not been loaded into the engine.")
+        new_pipeline_ids = []
+        pipelines = copy(pipelines)
+        while self._should_continue_callback() and len(pipelines) > 0:
+            pipeline = pipelines[0]
+            self._pre_evaluation_callback(pipeline)
+            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train)
+            new_pipeline_ids.append(self._post_evaluation_callback(pipeline, evaluation_result))
+            pipelines.pop(0)
+        return new_pipeline_ids
```

Comment:
A weakness of this code: if `train_and_score_pipeline` or one of the callbacks errors out, the whole `while` loop would halt.

We use the error callback inside `train_and_score_pipeline` to handle errors during pipeline evaluation, so this isn't a big concern. But if the data splitter errors, it would move up the stack. I think that behavior is acceptable for now.

Reply:
That's not dumb! You are correct, adding proper error handling now is the responsible thing to do. I was being lazy. Will do.

Reply:
I took a close look at this. I still agree that we should update the way we do error handling. But I'd like us to do it in a separate PR. There's a decent amount to consider. This PR doesn't change the existing error handling pattern; we will have to in order to complete the parallel work.